handler/security-handler-saml/src/main/java/org/codice/ddf/security/handler/saml/SAMLAssertionHandler.java

```diff
@@ -90,32 +88,44 @@ public HandlerResult getNormalizedToken(ServletRequest request, ServletResponse
                 LOGGER.trace("Cookie value: {}", tokenString);
                 securityToken = new SecurityToken();
                 Element thisToken = StaxUtils.read(new StringReader(tokenString))
-                  .getDocumentElement();
+                        .getDocumentElement();
                 securityToken.setToken(thisToken);
-                SAMLAuthenticationToken samlToken = new SAMLAuthenticationToken(null, securityToken, realm);
+                SAMLAuthenticationToken samlToken = new SAMLAuthenticationToken(null, securityToken,
+                        realm);
                 handlerResult.setToken(samlToken);
                 handlerResult.setStatus(HandlerResult.Status.COMPLETED);
             } catch (DataFormatException e) {
-                LOGGER.warn("Unexpected error deflating cookie value - proceeding without SAML token.", e);
+                LOGGER.warn(
+                        "Unexpected error deflating cookie value - proceeding without SAML token.",
+                        e);
             } catch (Base64Exception e) {
-                LOGGER.warn("Unexpected error un-encoding the cookie value - proceeding without SAML token.", e);
+                LOGGER.warn(
+                        "Unexpected error un-encoding the cookie value - proceeding without SAML token.",
+                        e);
             } catch (IOException e) {
-                LOGGER.warn("Unexpected error converting cookie value to string - proceeding without SAML token.", e);
+                LOGGER.warn(
+                        "Unexpected error converting cookie value to string - proceeding without SAML token.",
+                        e);
             } catch (XMLStreamException e) {
-                LOGGER.warn("Unexpected error converting XML string to element - proceeding without SAML token.", e);
+                LOGGER.warn(
+                        "Unexpected error converting XML string to element - proceeding without SAML token.",
+                        e);
             }
             return handlerResult;
         }
 
         HttpSession session = httpRequest.getSession(false);
-        if(session != null) {
+        if (session == null && httpRequest.getRequestedSessionId() != null) {
```

Comment:
Yeah we were always creating a session before but found that Jetty's getSession() method (no boolean) was not actually threadsafe and would cause some issues if you called it on absolutely every request. Moving to getSession(false) appeared to resolve the bugginess that we were seeing, but in doing so we messed up how SSO works across the container. So this way it allows SSO to work again without calling that unsafe method over and over.